Neos.Flow/Classes/ObjectManagement/DependencyInjection/ProxyClassBuilder.php

```diff
@@ -466,7 +468,34 @@ public function buildPropertyInjectionCodeByString(Configuration $objectConfigur
             return $result;
         }
 
-        if ($propertyConfiguration->isLazyLoading() && $this->objectConfigurations[$propertyObjectName]->getScope() !== Configuration::SCOPE_PROTOTYPE) {
+        $propertyClassHasInterfaceWithConstructor = false;
+        $propertyClassHasFinalMethod = false;
+        if ($propertyClassName) {
+            foreach ((new ClassReflection($propertyClassName))->getInterfaceNames() as $interfaceName) {
+                if (method_exists($interfaceName, '__construct')) {
+                    $propertyClassHasInterfaceWithConstructor = true;
+                    break;
+                }
+            }
+            foreach (get_class_methods($propertyClassName) as $methodName) {
+                if ($this->reflectionService->isMethodFinal($propertyClassName, $methodName)) {
+                    $propertyClassHasFinalMethod = true;
+                    break;
+                }
+            }
+        }
+
+        # There are several cases when providing lazy loading is not possible, due to the way
+        # lazy loading is implemented. If any of the check fails, we fall back to eager loading:
+        if (
+            $propertyConfiguration->isLazyLoading() &&
+            $this->objectConfigurations[$propertyObjectName]->getScope() !== Configuration::SCOPE_PROTOTYPE &&
```

Comment:
We dont allow prototypes lazly injected - but on my pc there are tousands of lazy proxies for prototypes generated ... isnt this a waste of performance?